lib/resources/host.rb

```diff
@@ -156,7 +160,8 @@ def ping(hostname, port, protocol)
 
       {
         success: resp.exit_status.to_i.zero?,
-        output: resp.stderr,
+        connection_output: resp.stderr,
```

Comment:
As stated in https://github.com/chef/inspec/pull/1927#issuecomment-308794898.  Netcat's `stdout` and `stderr` is what we're using now, but if we iterate on this in the future and find a tool that doesn't use `stdout`/`stderr` the same way, then we're training the user on a specific implementation.

There are two bits of data being exposed: output about the connection setup, and the output from the socket itself after the connection is established.

I proposed `connection_output` and `socket_output` accordingly, and I think the `_output` piece is important so someone doesn't believe these can be manipulated -- it's simply the output of the connection attempt. I'm totally fine with removing `_output` if it helps calm any concerns.